File: pset6/readability/readability.py
from cs50 import get_string

# getting sentence from user input
text = get_string('Text: ')

# using split() to count words in string
words = len(text.split())
# counting occurrencies of '.', '?' and '!' for number of sentences
sentences = text.count('!') + text.count('.') + text.count('?')
# counting all characters
letters = 0
for word in text.split():
    for c in word:
        if c >= 'A' and c <= 'z':
            letters += 1

L = letters / words * 100
S = sentences / words * 100

grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print('Before Grade 1\n')
elif grade > 16:
    print('Grade 16+\n')
else:
    print('Grade', grade)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int words = 1, letters = 0, sentences = 0;      // initializing words as 1 to count last word that isn't followed by a space
    string text = get_string("Text: ");             // getting text from user
    int len = strlen(text);                         // getting text length



    letters = len;

    for (int i = 0; i < len; i++)                   // iterating over text
    {
        if (text[i] == ' ')                         // counting words
        {
            words++;
            letters--;
        }
        else if (text[i] == '!' || text[i] == '.' || text[i] == '?')        // counting sentences
        {
            sentences++;
            letters--;
        }
        else if (text[i] == ',' || text[i] == '"' || text[i] == ':' || text[i] == '-' || text[i] == 39 || text[i] == ';')
        {
            // removing espacial characters from letters count
            letters--;

        }
    }

    float L = (float)letters / words * 100;               // L is the average number of letters per 100 words in the text
    float S = (float)sentences / words * 100;             // S is the average number of sentences per 100 words in the text

    float grade = round(0.0588 * L - 0.296 * S - 15.8);     // calculating grade

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", grade);
    }

}